File: baekjoon/04-DP/python/1912.py
import sys
sys.stdin = open('input.txt', 'r')

n = int(input())
arr = list(map(int, input().split()))
dp = [-1000 for _ in range(n)]

dp[0] = arr[0]
for i in range(1, n):
    dp[i] = max(dp[i-1] + arr[i], arr[i])

print(max(dp))

File: baekjoon/04-DP/js/1912.js
const fs = require("fs")
const input = fs.readFileSync('/dev/stdin').toString().split("\n")
const n = Number(input[0])
const seq = input[1].split(" ").map(Number)

let dp = new Array(n).fill(-1000)
dp[0] = seq[0]
for (let i = 1; i < n; i++) {
  dp[i] = Math.max(dp[i-1] + seq[i], seq[i])
}

console.log(Math.max(...dp))
